src/compiler/checker.ts

```diff
@@ -7852,16 +7842,18 @@ namespace ts {
                 case SyntaxKind.NeverKeyword:
                     return neverType;
                 case SyntaxKind.ObjectKeyword:
-                    return nonPrimitiveType;
+                    if (node.flags & NodeFlags.JavaScriptFile) {
+                        return anyType;

```

Comment:
There seems to be duplicate code in `getPrimitiveTypeFromJSDocTypeReference`, are these both needed?